### PR TITLE
feat(output): Enabling Configurable Outputter for Writing Proofs

### PR DESCRIPTION
**Pull request by envestcc**
Now users can customize the type of output in the project configuration. Three types are supported: 
- **stdout**: Print proof in stdout. It's the default output type if not configured
- **ethereumContract**: Write proof to an eth contract.
- **solanaProgram**: Write proof to a solana program

NOTE! There have been the following changes in the interface for calling ethereum contracts:
```setProof(bytes memory _proof)``` => ``` submitProof(bytes memory _proof)```

### CODE REVIEW

File: project/project.go [hunk 1]
```diff
@@ -10,9 +10,19 @@ type Project struct {
...
 type Config struct {
-	Code          string       `json:"code"`
-	CodeExpParam  string       `json:"codeExpParam,omitempty"`
-	VMType        types.VM     `json:"vmType"`
-	OutputType    types.Output `json:"outputType"`
-	OutputAddress string       `json:"outputAddress,omitempty"`
+	Code         string   `json:"code"`
+	CodeExpParam string   `json:"codeExpParam,omitempty"`
+	VMType       types.VM `json:"vmType"`
+	Output       struct {
```
Comment (huangzhiran) [inline project/project.go:16]: maybe define a OutputConfig struct would be cleaner

Comment (envestcc) [reply project/project.go:16]: > maybe define a OutputConfig struct would be cleaner

Okay

Comment (envestcc) [reply project/project.go:16]: > and also, the project config file in test/project dir need changed too

Actually `stdout` is used by default if output not configured, but we can also explicitly configure output as stdout

File: cmd/node/env.go [hunk 2]
```diff
@@ -1,14 +1,16 @@
...
-	DatabaseDSN            = "DATABASE_DSN"
-	Halo2ServerEndpoint    = "HALO2_SERVER_ENDPOINT"
-	OperatorPrivateKey     = "OPERATOR_PRIVATE_KEY"
-	ProjectContractAddress = "PROJECT_CONTRACT_ADDRESS"
-	ProjectFileDirectory   = "PROJECT_FILE_DIRECTORY"
-	Risc0ServerEndpoint    = "RISC0_SERVER_ENDPOINT"
-	ZkwasmServerEndpoint   = "ZKWASM_SERVER_ENDPOINT"
-	BootNodeMultiaddr      = "BOOTNODE_MULTIADDR"
-	IotexChainID           = "IOTEX_CHAINID"
+	ChainEndpoint             = "CHAIN_ENDPOINT"
```
[the remark this answers is not in the document]
Comment (envestcc) [reply cmd/node/env.go:4]: It's still used to init projectManager, but could consider to use chainConfig instead in another PR

**Review by huangzhiran — APPROVED** [review]

**Pull request merged**